### DISCUSSION TITLE
Save Passport Reminder

### DISCUSSION THREAD

**Discussion by bhoanged** [jdholtz/auto-southwest-check-in#182] [Ideas]
I always forget to save my passport information beforehand, would it be possible to add a command line message reminding if any leg of the flight is not in the US?

**Comment by jdholtz** [comment]
I like that idea. I'm currently looking at how to get the country from the flight information. For me, domesticated flights show the country as `None` but the state has a value. However, I'm not sure what it looks like for international flights. If you are comfortable sharing, could you put `print(response)` on line 92 of `lib/checkin_scheduler.py`? This shows all of the reservation information, but feel free the change the values (such as your name, the flight number, etc.) if you want to hide any info. One key that seems interesting to me is `isInternational`, which may be the solution to determining if a flight is international (if it is set correctly by Southwest).  

Also, do you only need to fill out passport information on the flight leaving the U.S or both the departing and returning flights?

**Reply by cmclinton** [reply]
Alternatively, could there be a lookup? Southwest only has a few international destinations:

https://www.southwest.com/route-map-list-view/?clk=RTMAPLIST

Aruba, Aruba - AUA
Providenciales, Turks and Caicos - PLS
Punta Cana, Dominican Republic - PUJ
Montego Bay, Jamacia - MBJ
Grand Cayman, Cayman Islands - GCM
Liberia, Costa Rica - LIR
San Jose, Costa Rica - SJO
Belize City, Belize - BZE
Cancun, Mexico - CUN
Cozumel, Quintana Roo - CZM
Puerto Vallarta, Mexico - PVR
Los Cabos, Mexico - SJD
Nassau, Bahamas - NAS
Havana, Cuba - HAV
San Juan, Puerto Rico - SJU (not sure if passport is needed)

**Reply by cmclinton** [reply]
Or If country value is populated (since domestic is blank) then do passport stuff

**Reply by jdholtz** [reply]
Yes, I could do that. Thanks for finding that list. However, I would prefer if it could be implemented just using the info from Southwest as that will be more robust to future changes.

**Comment by bhoanged** [comment]
It looks like this, I didn't see a isInternational key:
{'departureStatus': 'ON TIME', 'departureStatusType': 'POSITIVE', 'arrivalStatus': 'ON TIME', 'arrivalStatusType': 'POSITIVE', 'actualDepartureTime': '', 'actualArrivalTime': '', 'flights': [{'number': '', 'wifiOnBoard': True, 'aircraftInfo': {'aircraftType': 'Boeing 737-700', 'numberOfSeats': 143, 'wifiSupported': True}}], 'travelTime': '', 'departureDate': '', 'departureTime': '', 'departureAirport': {'name': '', 'code': '', 'country': None}, 'arrivalTime': '', 'arrivalAirport': {'name': 'Cancun', 'state': None, 'code': 'CUN', 'country': 'Mexico'}, 'passengerTypeCounts': {'adult': 1}, 'passengers': [{'type': 'Passenger', 'count': 1, 'fareType': 'Wanna Get Away'}], 'boundType': 'DEPARTING', 'standbyFlight': None, 'fareProductDetails': {'label': 'Wanna Get Away', 'fareRulesUrl': 'https://mobile.southwest.com/fare-rules/wanna-get-away', 'fareProductId': 'WGA'}, 'stops': [], 'isNextDayArrival': False} 

I'm 90% sure that you only need to fill out on the flight leaving the U.S. I'll be able to know for sure once I check in for my return next week.

**Reply by jdholtz** [reply]
Sorry, I edited the message to show the reservation information instead of just the flight info. The country is populated so I may be able to use that.

**Comment by jdholtz** [comment]
Added in the latest develop commit. Thanks for the feature request!
